Unity-Technologies/com.unity.multiplayer.samples.coop
# fix: client not properly disconnecting when host disconnects, quits or crashes

### Description (*)

This PR removes the `!IsConnectedClient` condition for handling client disconnect. This is because the latest version of Netcode changed the order of calling `OnClientDisconnectCallback` and `Shutdown` (see [this PR](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/1221)). The callback is now called before shutting down, which means that `IsConnectedClient` still returns true.
### Related Pull Requests

### Issue Number(s) (*)

Fixes issue(s): [MTT-1849](https://jira.unity3d.com/browse/MTT-1849), [MTT-1971](https://jira.unity3d.com/browse/MTT-1971), and [MTT-1986](https://jira.unity3d.com/browse/MTT-1986)
### Manual testing scenarios

1. Start a game as a host
2. Join as a client
3. Quit the game as the host
4. See the client return to main menu correctly

To test for hard crashes, you can replace line 60 in `ServerTestingHotkeys.cs` with `Utils.ForceCrash(ForcedCrashCategory.FatalError);`, then press `q` on the host while in-game to force a crash.
### Questions or comments

### Contribution checklist
